src/metrics/utils/multiaddress.go

```diff
@@ -0,0 +1,74 @@
+package utils
+
+import (
+	"fmt"
+	"net"
+	"strings"
+
+	"github.com/pkg/errors"
+)
+
+// Get the Real Ip Address from the multi Address list
+// TODO: -- IP could be returned after filtering it
+func GetFullAddress(multiAddrs []string) (string, error) {
+	// TODO: It's pretty unlikely that we don't have the multiaddress of a discovered peer
+	//       however, should we leave it empty if we don't have it? All the fields where filled
+	//       before for the post analysis in the Python Script, does it still make sense?
+	var address string = "/ip4/127.0.0.1/tcp/9000"
+	if len(multiAddrs) > 0 {
+		fmt.Println(multiAddrs)
+		address = multiAddrs[0]
+		// TODO: Missing net.IP from multiaddress
+		for _, addrs := range multiAddrs {
+			// parse IP from Addrs
+			ip, err := GetIpFromMultiAddr(addrs)
+			if err != nil {
+				continue
+			}
+			if IsPublic(ip) {
+				address = addrs
+				break
+			} else {
+				continue
+			}
+		}
+		return address, nil
+	}
+	return address, errors.New("Given set of multiAddresses is empty")
+}
+
+// Extract IP from Multiaddress
+func GetIpFromMultiAddr(addrs string) (net.IP, error) {
```

Comment:
Perhaps we can reuse this library `go-multiaddr` ? Rumor uses it and I believe we can use it to get the remote IP.
https://github.com/multiformats/go-multiaddr/blob/7cb054e4eed3c33c2708fd58c4816db652d14e26/net/net.go#L26-L28